### PR TITLE
Fix count issue

### PR DESCRIPTION
**Pull request by leiserfg**
- change mapping into function-based interface
- Pass counter in normal call
- Replace count with count1

### CODE REVIEW

**Comment by leiserfg** [comment]
This fixes the the issue with count, I also replaced the other usages of `vim.v.count` I can remove the commit if it feels inappropriate to you.

File: lua/dial/map.lua [hunk 1]
```diff
@@ -25,7 +25,7 @@ local function _cmd_sequence(direction, mode, group_name)
     -- command.select_augend_normal(vim.v.count, group_name)
     local setopfunc = cmdcr([[let &opfunc="dial#operator#]] .. direction .. "_" .. mode .. [["]])
     local textobj = util.if_expr(mode == "normal", cmdcr [[lua require("dial.command").textobj()]], "")
-    return select .. setopfunc .. "g@" .. textobj
+    return vim.v.count1..select .. setopfunc .. "g@" .. textobj
```
Comment (monaqa) [inline lua/dial/map.lua:28]: In my environment it doesn't work, but by making the following changes, it works. How do you like it?

```suggestion
    return select .. setopfunc .. vim.v.count1 .. "g@" .. textobj
```

Comment (leiserfg) [reply lua/dial/map.lua]: I applied your suggestion, anything else I should do to get this merged?

**Comment by monaqa** [comment]
Sorry for the late reply, I will merge.

**Pull request merged**